### PR TITLE
HOWTO Add instruction for using the aws-sam-cli, It is a wrapper for cloud …

### PR DESCRIPTION
**Pull request by janitha09**
…formation

*Issue #, if available:*

*Description of changes:*
Exploring the idea of providing a link out to the aws-sam-cli

*Description of how you validated changes:*

*Checklist:*

- [ ] Write/update tests
- [ ] `make pr` passes
- [x] Update documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: HOWTO.md [hunk 1]
```diff
@@ -14,9 +14,15 @@ resources for you.
 The remainder of this document explains how to write SAM templates and 
 deploy them via AWS CloudFormation. 
 
-## Writing SAM Template
+## Getting started with SAM Template
 Check out the [latest specification](versions/2016-10-31.md) for details on how to write a SAM template
 
+
+You could also use the [aws-sam-cli](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-template-publishing-applications.html) to get started
```
Comment (michaelj-smith) [inline HOWTO.md]: As a GitHub readme file, it's probably more direct to link to the GitHub repo for SAM CLI:
https://github.com/awslabs/aws-sam-cli

File: HOWTO.md [hunk 2]
```diff
@@ -50,6 +56,15 @@ $ aws cloudformation package \
     --output-template-file packaged-template.yaml
 ```
 
+Using `aws-sam-cli`
+
+```bash
+$ sam package \
+    --template-file /path_to_template/template.yaml \
+    --s3-bucket bucket-name \
+    --output-template-file packaged-template.yaml
```
Comment (michaelj-smith) [inline HOWTO.md]: This was just today being discussed in #830. The command for `aws cloudformation package` has the additional option of specifying --s3-prefix, which is recommended if you're going to be using S3 as an artifact repository.

File: HOWTO.md [hunk 3]
```diff
@@ -80,7 +95,16 @@ $ aws cloudformation deploy \
...
 ```
 
-Refer to the [documentation](http://docs.aws.amazon.com/cli/latest/reference/cloudformation/deploy/index.html) for more details.
+Or using `aws-sam-cli` 
+
+```bash
+$ sam publish \
+    --template-file /path_to_template/packaged-template.yaml \
+    --region AWS_REGION
+    --capabilities CAPABILITY_IAM
```
Comment (michaelj-smith) [inline HOWTO.md]: The `sam publish` command is VERY different than the `sam deploy` command. Best not to get those mixed up in the basic how-to documentation.
`sam deploy` is an alias for `aws cloudformation deploy`.
`sam publish` publishes the app to the Serverless Application Repository.

File: HOWTO.md (hunk 2 above)
Comment (janitha09) [reply HOWTO.md]: Sam most certainly reduces the amount of typing one has to do. sam init is fantastic. However some parts aren't necessarily very user friendly. For example sam package help 

```shell 
$ sam --version
SAM CLI, version 0.12.0
$ sam package help
Usage: sam package [OPTIONS] [ARGS]...

Error: Missing option "--s3-bucket".
```

Which is very different to the documentation brought up by aws cloudformation package help. But that takes over window, which is maybe (haven't tried) a problem for popen?

**Pull request merged**